src/Cli/dotnet/Parser.cs

```diff
@@ -217,7 +190,7 @@ public override void Write(HelpContext context)
                 }
                 else if (command.Name.Equals(NewCommandParser.GetCommand().Name))
                 {
-                    NewCommandShim.Run(context.ParseResult.GetArguments());
+                    NewCommandShim.Run(helpArgs);
                 }
```

Comment:
This whole range can be deleted due to https://github.com/dotnet/sdk/pull/23380/files#r783164794

```suggestion
```